dissect/target/plugins/apps/chat/msn.py

```diff
@@ -0,0 +1,116 @@
+
+    @export(record=[ChatMessageRecord, ChatAttachmentRecord])
+    def history(self) -> Iterator[ChatMessageRecord | ChatAttachmentRecord]:
+        """Yield MSN chat history messages
+
+        Chat history artifacts can be found in:
+            - ``$HOME/My Documents/My Received Files/MsnMsgr.txt``
+            - ``$HOME/My Documents/My Received Files/$username$PassportID/History/*.xml``
+        """
+
+        for user_details, profile in self.installs:
+            if not (hist_root := self.target.fs.path(user_details.user.home).joinpath(self.HIST_PATH)).exists():
+                self.target.log.warning(
+                    "User %s does not have saved MSN chat history: directory %s does not exist",
+                    user_details.user.name,
+                    hist_root,
+                )
+                continue
+
+            hist_dir = None
+            for item in hist_root.iterdir():
+                if item.is_dir() and (hist_dir := item.name).endswith(profile.name):
+                    for hist_file in hist_root.joinpath(hist_dir).joinpath("History").glob("*.xml"):
+                        try:
+                            xml = ET.fromstring(hist_file.read_text())
+                        except Exception as e:
+                            self.target.log.warning("XML file %s is malformed: %s", hist_file, e)
+                            continue
+
+                        for entry in xml:
+                            common = {
+                                "ts": entry.attrib.get("DateTime", 0),
+                                "client": self.__namespace__,
+                                "account": profile.name,
+                                "sender": entry.find(".//From/User").get("FriendlyName"),
+                                "_user": user_details.user,
+                                "_target": self.target,
+                            }
+
+                            if entry.tag == "Message":
+                                yield ChatMessageRecord(
+                                    **common,
+                                    recipient=entry.find(".//To/User").get("FriendlyName"),
+                                    message=entry.find(".//Text").text,
+                                )
+
+                            elif entry.tag in ["Invitation", "InvitationResponse"]:
+                                if (file := entry.find(".//File")) is not None:
+                                    yield ChatAttachmentRecord(
+                                        **common,
+                                        recipient=None,  # unknown with Invitations
+                                        attachment=file.text,
+                                        description=entry.find(".//Text").text,
+                                    )
+
+
+def convert_email(string: str) -> int:
+    """Convert MSN email address to 10 digit Passport ID"""
```

Comment:
```suggestion
    """Convert MSN email address to 10 digit Passport ID."""
```
Can you also add a docstring what this is used for? From the test file I gather it's for the history path.